src/cow-react/modules/limitOrders/pure/Orders/OrderRow/index.tsx

```diff
@@ -198,12 +202,23 @@ export function OrderRow({
   // const executedTimeAgo = useTimeAgo(expirationTime, TIME_AGO_UPDATE_INTERVAL)
   const activityUrl = chainId && activityId ? getEtherscanLink(chainId, activityId, 'transaction') : undefined
 
-  const executionPriceInversed = isRateInversed
-    ? prices?.estimatedExecutionPrice.invert()
-    : prices?.estimatedExecutionPrice
-  // const marketPriceInversed = isRateInversed ? prices?.marketPrice.invert() : prices?.marketPrice
-  const executedPriceInversed = isRateInversed ? executedPrice?.invert() : executedPrice
-  const spotPriceInverted = isRateInversed ? spotPrice?.invert() : spotPrice
+  const [isInverted, setIsinverted] = useState(isRateInverted)
+
+  // Update internal isInverted flag whenever prop change
+  useEffect(() => {
+    setIsinverted(isRateInverted)
+  }, [isRateInverted])
+
+  // On mount, apply smart quote selection
+  useEffect(() => {
+    const quoteCurrency = getQuoteCurrency(chainId, inputCurrencyAmount, outputCurrencyAmount)
+    setIsinverted(getAddress(quoteCurrency) !== getAddress(inputCurrencyAmount?.currency))
+    // Intentionally empty, should run only once
+  }, [])
+
+  const executionPriceInverted = isInverted ? estimatedExecutionPrice?.invert() : estimatedExecutionPrice
+  const executedPriceInverted = isInverted ? executedPrice?.invert() : executedPrice
+  const spotPriceInverted = isInverted ? spotPrice?.invert() : spotPrice
```

Comment:
This is the actual fix.

The rest is noise from the typo.

If it helps, the fix in in this commit a86a9c7ea28fee044d77b23ccdd4c4ed583eeec5